python/ray/experimental/state.py

```diff
@@ -12,3 +54,198 @@ def get_local_schedulers(worker):
     if client_info[b"client_type"] == b"local_scheduler":
       local_schedulers.append(client_info)
   return local_schedulers
+
+
+class GlobalState(object):
+  """A class used to interface with the Ray control state.
+
+  Attributes:
+    redis_client: The redis client used to query the redis server.
+  """
+  def __init__(self):
+    """Create a GlobalState object."""
+    self.redis_client = None
+
+  def _check_connected(self):
+    """Check that the object has been initialized before it is used.
+
+    Raises:
+      Exception: An exception is raised if ray.init() has not been called yet.
+    """
+    if self.redis_client is None:
+      raise Exception("The ray.global_state API cannot be used before "
+                      "ray.init has been called.")
+
+  def _initialize_global_state(self, redis_ip_address, redis_port):
```

Comment:
I think this is tough because of the way we do `from ray.worker import global_state` in `__init__.py`, the problem is described here http://stackoverflow.com/questions/4758562/modifying-a-variable-in-a-module-imported-using-from-import.

That is, I think we can't easily reassign the variable after ray is imported, so we have to construct it the first time, but then we can't actually initialize the object until `ray.init` is called.